tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs

```diff
@@ -22,10 +23,42 @@ public static string GetFileOrUrlContents(string fileOrUrl)
 
         private static string GetUrlContents(string url)
         {
+            int maxRetries = 3;
+            int attempts = 1;
+            int delayTimeInMs = 1000;
             using HttpClient client = new HttpClient();
-            HttpResponseMessage response =
-                client.GetAsync(url).ConfigureAwait(false).GetAwaiter().GetResult();
-            return response.Content.ReadAsStringAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+            while (attempts <= maxRetries)
+            {
+                try
+                {
+                    HttpResponseMessage response = client.GetAsync(url).ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
I don't know how many changes it would require in the codeowners tool, but in the future it would be nice to just make this whole function async. Not sure why it isn't.

```
                    HttpResponseMessage response = await client.GetAsync(url);
```

Reply:
I agree. It's most likely the fact that making this one method async would cascade through the chain of calling functions that would require things using Codeownes to be async that wouldn't normally be.